### PR TITLE
initial commit with obsidian interactive graph feature

### PR DESCRIPTION
**Pull request by daxcore**
maybe not perfect yet... but IMO a good starting point if u 're interested into this feature?!

more information here:
https://echarts.apache.org/en/index.html

demo and live testing:
https://echarts.apache.org/examples/en/editor.html?c=graph

### CODE REVIEW

**Comment by daxcore** [comment]
dark:

normal:

mouse over:
test

File: docs/assets/javascript/graph.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  });
+  
+  var dom = document.getElementById('graph');
+  var myChart = echarts.init(dom, null, {
+    renderer: 'canvas',
+    useDirtyRect: true
+  });
+  var option;
+  
+  $.getJSON(window.location.origin + '/data.json', function (graph) {
```
Comment (daxcore) [inline docs/assets/javascript/graph.js:13]: the data.json file will be generated by the python hook.

File: docs/assets/javascript/graph.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    myChart.hideLoading();
+    graph.nodes.forEach(function (node) {
+      node.symbolSize += 5;
```
Comment (daxcore) [inline docs/assets/javascript/graph.js:16]: min size of the nodes in the graph is 5. 
the size will be increased be each edge of the node by 1.

File: docs/assets/javascript/graph.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    });  
+    option = {
+      tooltip: {
+        show: false,
+        formatter: '<b>{b0}</b>'
+      },
+      legend: [
```
Comment (daxcore) [inline docs/assets/javascript/graph.js:23]: i do not use categories for now.

File: docs/assets/javascript/graph.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    });
+  });
+  
+  if(option && typeof option === 'object') {
+    myChart.setOption(option);
+  }
+  
+  window.addEventListener('resize', myChart.resize);
+  
+  $("#__palette_0").change(function(){
```
Comment (daxcore) [inline docs/assets/javascript/graph.js:84]: handle changing between light and dark theme.

File: docs/assets/javascript/graph.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+          lineStyle: {
+            color: 'source',
+            curveness: 0 //0.3
+          }
+        }
+      ]
+    };
+  
+    myChart.setOption(option);
+    myChart.on('click', function (params) {
```
Comment (daxcore) [inline docs/assets/javascript/graph.js:70]: make the nodes clickable and link with correct markdown site.

**Comment by ndy2** [comment]
Thank you for suggesting new features and providing drafts.
However, it feels unclear what this feature is trying to achieve.

Personally, while using Obsidian, I only used the traditional dictionary and tag method rather than graph to organize documents. I am curious as to whether you want to place the generated graph on the sidebar like in https://help.obsidian.md/Home.

**Comment by daxcore** [comment]
refer https://daxcore.github.io/mkdocs-obsidian-interactive-graph/

**Pull request closed without merging**